lib/kafka_rest_client.js

```diff
@@ -284,6 +284,12 @@ KafkaRestClient.prototype.tryMakeRequest = function tryMakeRequest(produceMessag
                 if (self.statsd) {
                     self.statsd.increment(metricPrefix + '.retry');
                 }
+            } else if (self.isHeatPipeTopic(produceMessage.topic) && retry === self.maxRetries - 1 && self.maxRetries > 1) {
+                /* eslint-disable no-undef,block-scoped-var */
+                setTimeout(function delayedProduce() {
+                    self.tryMakeRequest(produceMessage, self.maxRetries, callback);
```

Comment:
I think it's better to directly retry here instead of call tryMakeRequest again.